lib/filterx/tests/test_object_json.c

```diff
@@ -103,35 +103,27 @@ Test(filterx_json, test_json_function)
   assert_object_json_equals(fobj, "{\"foo\":1}");
   filterx_object_unref(fobj);
 
+  /* message_value is being unmarshalled before passing it to a simple
+   * function, so no need to handle it */
   fobj = _exec_json_func(filterx_message_value_new("{\"foo\": 1}", -1, LM_VT_JSON));
-  cr_assert(filterx_object_is_type(fobj, &FILTERX_TYPE_NAME(json_object)));
-  assert_object_json_equals(fobj, "{\"foo\":1}");
-  filterx_object_unref(fobj);
+  cr_assert(fobj == NULL);
 
   fobj = _exec_json_func(filterx_string_new("[1, 2]", -1));
   cr_assert(filterx_object_is_type(fobj, &FILTERX_TYPE_NAME(json_array)));
   assert_object_json_equals(fobj, "[1,2]");
   filterx_object_unref(fobj);
 
   fobj = _exec_json_func(filterx_message_value_new("[1, 2]", -1, LM_VT_JSON));
-  cr_assert(filterx_object_is_type(fobj, &FILTERX_TYPE_NAME(json_array)));
-  assert_object_json_equals(fobj, "[1,2]");
-  filterx_object_unref(fobj);
+  cr_assert(fobj == NULL);
 
   fobj = _exec_json_func(filterx_message_value_new("foo,bar", -1, LM_VT_LIST));
-  cr_assert(filterx_object_is_type(fobj, &FILTERX_TYPE_NAME(json_array)));
-  assert_object_json_equals(fobj, "[\"foo\",\"bar\"]");
-  filterx_object_unref(fobj);
+  cr_assert(fobj == NULL);
 
   fobj = _exec_json_func(filterx_message_value_new("{\"foo\": 1}", -1, LM_VT_STRING));
-  cr_assert(filterx_object_is_type(fobj, &FILTERX_TYPE_NAME(json_object)));
-  assert_object_json_equals(fobj, "{\"foo\":1}");
-  filterx_object_unref(fobj);
+  cr_assert(fobj == NULL);
 
   fobj = _exec_json_func(filterx_message_value_new("[1, 2]", -1, LM_VT_STRING));
-  cr_assert(filterx_object_is_type(fobj, &FILTERX_TYPE_NAME(json_array)));
-  assert_object_json_equals(fobj, "[1,2]");
-  filterx_object_unref(fobj);
+  cr_assert(fobj == NULL);
```

Comment:
It looks like these test cases became useless as they try to test something that is "not supported" on the given interface. Can we keep the first one to document the behavior and get rid of the rest?